models/user.go

```diff
@@ -317,13 +334,7 @@ func (u *User) NewGitSig() *git.Signature {
 
 // EncodePasswd encodes password to safe format.
 func (u *User) EncodePasswd() {
-	var hashAlgorithm func() hash.Hash
-	switch setting.Cfg.Section("security").Key("PASSWORD_HASH_ALGORITHM").String() {
-	case "sha512":
-		hashAlgorithm = sha512.New
-	default:
-		hashAlgorithm = sha256.New
-	}
+	updateHashAlgorithm()
```

Comment:
This still runs once on each password-change... see this for reference: https://github.com/gogits/gogs/pull/2406/files